### DISCUSSION TITLE
GetIt Package Manager installer

### DISCUSSION THREAD

**Discussion by veftodii** [graphics32/graphics32#254] [General] [answered]
Is this library available for installation through GetIt Package Manager? I searched and didn't find it. I tried Delphi 10.4.2 and 11.3.
I was wondering if this may be easier to handle than downloading, building, and installing manually. For example, I tried "Abbrevia" to see how this works.
The most interesting thing is that I can add this library as a required dependency in project options. Though this has its disadvantages too as new library releases are not updated between Delphi versions sometimes.

**Comment by andersmelander** [comment, marked as answer]
> Is this library available for installation through GetIt Package Manager?

No. See #251

> I was wondering if this may be easier to handle than downloading, building, and installing manually.

Yes, it would be easier but the disadvantage is that GetIt isn't a very dynamic system. For example, I wouldn't be able to push new releases and have them made available through GetIt immediately (or even after a short time). The release schedule is purely controlled by Embarcadero.

Also, Embarcadero has absolutely no interest in maintaining support for older versions of Delphi, so it would only be the latest version of Delphi that got updates through GetIt. Older versions of Delphi would be stuck with the last version that was available when the new version of Delphi was released. If you have a 10.x version of Delphi installed you can see that the GetIt content for it isn't updated anymore.
In other words: GetIt is mainly a marketing tool. It isn't meant to make things easier for third-parties or even the users.

**Reply by veftodii** [reply]
I was thinking that I imagined things but you encountered them too. This makes the GetIt Package Manager somehow useless. For example, "Abbrevia" is also on GitHub. In Delphi 10.4.2 you download one old version and in 11.3 the latest version. If I enable this dependency in project options then I can't have the same settings in D10 and D11. I need to modify this setting any time I open the project in one of these Delphi versions, which makes the entire process just simply stupid. So, I think I will download "Abbrevia" from GitHub and install it manually without any dependency settings in the project options. I did this for Graphics32 and it is workings as it needs to be.
Thank you for your thoughts on this matter and your explanation!

**Reply by micha137** [reply]
> Yes, it would be easier but the disadvantage is that GetIt isn't a very dynamic system. For example, I wouldn't be able to push new releases and have them made available through GetIt immediately (or even after a short time). The release schedule is purely controlled by Embarcadero.
> 
> Also, Embarcadero has absolutely no interest in maintaining support for older versions of Delphi, so it would only be the latest version of Delphi that got updates through GetIt. Older versions of Delphi would be stuck with the last version that was available when the new version of Delphi was released. If you have a 10.x version of Delphi installed you can see that the GetIt content for it isn't updated anymore. In other words: GetIt is mainly a marketing tool. It isn't meant to make things easier for third-parties or even the users.

Very well said.  I also had this on my todo list but never knew why its priority stayed so low.  Now I will strike it.

A better alternative is [Delphinus](https://github.com/Memnarch/Delphinus/wiki) - where I have already made Gr32 availble for (but didn't check for the state for long).
